types/object.go

```diff
@@ -152,6 +152,126 @@ func (t ObjectType) ValueType(_ context.Context) attr.Value {
 	}
 }
 
+// ObjectNull creates a Object with a null value. Determine whether the value is
+// null via the Object type IsNull method.
+//
+// Setting the deprecated Object type AttrTypes, Attrs, Null, or Unknown fields
+// after creating a Object with this function has no effect.
+func ObjectNull(attributeTypes map[string]attr.Type) Object {
+	return Object{
+		attributeTypes: attributeTypes,
+		state:          valueStateNull,
+	}
+}
+
+// ObjectUnknown creates a Object with an unknown value. Determine whether the
+// value is unknown via the Object type IsUnknown method.
+//
+// Setting the deprecated Object type AttrTypes, Attrs, Null, or Unknown fields
+// after creating a Object with this function has no effect.
+func ObjectUnknown(attributeTypes map[string]attr.Type) Object {
+	return Object{
+		attributeTypes: attributeTypes,
+		state:          valueStateUnknown,
+	}
+}
+
+// ObjectValue creates a Object with a known value. Access the value via the Object
+// type ElementsAs method.
+//
+// Setting the deprecated Object type AttrTypes, Attrs, Null, or Unknown fields
+// after creating a Object with this function has no effect.
+func ObjectValue(attributeTypes map[string]attr.Type, attributes map[string]attr.Value) (Object, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	// Ideally, Type() would not require a context.Context as it has no benefit
+	// here or elsewhere. There is also no benefit to adding it to the function
+	// parameters at the moment.
+	ctx := context.Background()
```

Comment:
I'll raise an issue, sorry for the code comment banter here. We can decide whether its a good idea or not (I personally think so because the type information should be very static and not need anything like logging, but maybe I'm off base here).